gax-grpc/src/test/java/com/google/api/gax/grpc/GrpcCallableFactoryTest.java

```diff
@@ -88,7 +88,10 @@ public void createServerStreamingCallableRetryableExceptions() throws Exception
     ServerStreamingCallSettings<Color, Money> nonRetryableSettings =
         ServerStreamingCallSettings.<Color, Money>newBuilder()
             .setRetrySettings(
-                RetrySettings.newBuilder().setTotalTimeout(Duration.ofSeconds(1)).build())
+                RetrySettings.newBuilder()
+                    .setTotalTimeout(Duration.ofSeconds(1))
+                    .setMaxAttempts(1)
+                    .build())
```

Comment:
How exactly does this fix work?

Reply:
The test doesn't actually want retry attempts, it just needs to ensure that the ApiException is retryable. Before setting `maxAttempts`, the retry logic would keep retrying for 1 second. However this mingled poorly with the fact that `ServerStreamingAttemptCallable` uses the `totalTimeout` as the grpc deadline. Depending on timer ordering either the retry mechanism would detect that `totalTimeout` has been reached and go down the expected path and throw the last error, which was `InvalidArgumentException`. Or the request would go out to grpc as the timeout is being reached and grpc would kill the request with an unexpected deadline exceeded error, causing the test to fail

Reply:
Ok sounds good!